gradle-jdks/src/main/java/com/palantir/gradle/jdks/GradleJdksConfigs.java

```diff
@@ -48,6 +52,13 @@ public abstract class GradleJdksConfigs extends DefaultTask {
     public static final String GRADLE_JDKS_SETUP_SCRIPT = "gradle-jdks-setup.sh";
     public static final String GRADLE_JDKS_FUNCTIONS_SCRIPT = "gradle-jdks-functions.sh";
 
+    @Inject
+    protected abstract ObjectFactory getObjectFactory();
+
+    private GradleOperatingSystem gradleOperatingSystem = getObjectFactory().newInstance(GradleOperatingSystem.class);
+
+    private final Provider<OperatingSystem> operatingSystem = gradleOperatingSystem.getOperatingSystem();
```

Comment:
Can we make a `@Internal @Nested GradleOperatingSystem` instead?

Reply:
Yep, Ill make it nested, now that https://github.com/palantir/gradle-utils/pull/397 is out